library/Makefile

```diff
@@ -11,6 +10,9 @@ LOCAL_LDFLAGS =
 
 ifdef DEBUG
 LOCAL_CFLAGS += -g3
+CFLAGS	?= -O0
```

Comment:
Good point, I'll remove this commit from the PR.

Reply:
I'd prefer `DEBUG` to set `-O0` but it doesn't belong in this PR.

Reply:
@gilles-peskine-arm: Interesting, why is that? IMHO it only causes problems when debugging because the compiled code is extremely different between -O2 and -O0. Also, I am not sure people rely on -O0 much because the compiled code is really poor quality.

Reply:
I personally prefer `-O0` because most of the bugs are present there as well and it allows to debug the C source instead of raw assembly. Of course in some nasty cases the bug is not reproducible with `-O0`, but so far those have been very rare.

Reply:
If you do printf debugging then there's no point in changing `-O` when compiling for debugging. But we don't do that except sometimes for TLS. When you're using a debugger, `-O0` is a lot easier to work with: stepping in code follows the code, breakpoints trigger when you expect them… I can't remember a single bug in Mbed TLS that went away with `-O0` and that Asan or Valgrind didn't reveal.